test/functional/output_kafka.go

```diff
@@ -0,0 +1,102 @@
+package functional
+
+import (
+	"github.com/ViaQ/logerr/log"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/helpers/kafka"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+	"strconv"
+	"strings"
+)
+
+const (
+	// Kafka deployment definitions
+	ImageRemoteKafka            = kafka.KafkaImage
+	ImageRemoteKafkaInit        = kafka.KafkaInitUtilsImage
+	OpenshiftLoggingNS          = "openshift-logging"
+	kafkaBrokerContainerName    = "broker"
+	kafkaBrokerComponent        = "kafka"
+	kafkaBrokerProvider         = "openshift"
+	kafkaNodeReader             = "kafka-node-reader"
+	kafkaNodeReaderBinding      = "kafka-node-reader-binding"
+	kafkaInsidePort             = int32(9093)
+	kafkaOutsidePort            = int32(9094)
+	kafkaJMXPort                = int32(5555)
+	zookeeperClientPort         = int32(2181)
+	zookeeperPeerPort           = int32(2888)
+	zookeeperLeaderElectionPort = int32(3888)
+)
+
+func (f *FluentdFunctionalFramework) addKafkaOutput(b *runtime.PodBuilder, output logging.OutputSpec) error {
```

Comment:
@pmoogi-redhat you are using two different "creation" approaches in this function (a) and (b) directly using the builder and (c) using the "old" helper ... I suggest moving to one of the options ... either rely on the "old" way of creating objects like (c) or more to new ... haveing a mix will be support hell in the future.